### PR TITLE
add esquery to history

### PR DESCRIPTION
**Pull request by 31453**
only implemented for sessions queries

### CODE REVIEW

File: db/db.pl [hunk 1]
```diff
@@ -5237,6 +5238,11 @@ sub historyUpdate
       },
       "forcedExpression": {
         "type": "keyword"
+      },
+      "esQuery": {
```
Comment (31453) [inline db/db.pl:5242]: should we stringify the json instead?

File: db/db.pl [hunk 2]
```diff
@@ -7311,7 +7317,7 @@ sub dataNodes
         sessions3Update();
         historyUpdate();
         fieldsUpdate();
-    } elsif ($main::versionNumber <= 71) {
+    } elsif ($main::versionNumber <= 72) {
```
Comment (31453) [inline db/db.pl:7320]: need this above (in the < 71 part) too?

File: viewer/viewer.js [hunk 3]
```diff
@@ -17,7 +17,7 @@
  */
 'use strict';
 
-const MIN_DB_VERSION = 70;
+const MIN_DB_VERSION = 72;
```
Comment (31453) [inline viewer/viewer.js:20]: we need to upgrade the version or else the history index will complain about the esquery field

File: viewer/viewerUtils.js [hunk 4]
```diff
@@ -273,8 +273,10 @@ module.exports = (Config, Db, molochparser, internals) => {
 
       const indices = await Db.getIndices(reqQuery.startTime, reqQuery.stopTime, reqQuery.bounding, Config.get('rotateIndex', 'daily'));
       if (indices.length > 3000) { // Will url be too long
+        req._molochESQuery = query;
```
Comment (31453) [inline viewer/viewerUtils.js:276]: Do you think this is the best place to do this? It's the last step buildSessionQuery. Alternatively, I could do it in the finish callback every time buildSessionQuery is called (~14 times).

File: viewer/vueapp/src/components/history/History.vue [hunk 5]
```diff
@@ -299,6 +299,11 @@
                     </em>
                   </div>
                 </div> <!-- /query params -->
+                <!-- es query -->
+                <div class="mt-3" v-if="item.esQuery">
+                  <h5>Elasticsearch Query</h5>
+                  <pre class="mr-3 ml-3">{{ item.esQuery }}</pre>
```
Comment (31453) [inline viewer/vueapp/src/components/history/History.vue:305]: Want this formatted in any special way?

File: db/db.pl (hunk 1 above)
Comment (awick) [reply db/db.pl:5242]: yes i think we should store as a string

File: db/db.pl (hunk 2 above)
Comment (awick) [reply db/db.pl:7320]: not in this case

File: viewer/viewer.js (hunk 3 above)
Comment (31453) [reply viewer/viewer.js:20]: leave it to 70 for now

File: viewer/vueapp/src/components/history/History.vue (hunk 5 above)
Comment (awick) [reply viewer/vueapp/src/components/history/History.vue]: MAKE PRETTY

**Pull request merged**